comp/core/gui/guiimpl/views/private/js/javascript.js

```diff
@@ -18,11 +18,62 @@ function sendMessage(endpoint, data, method, callback, callbackErr){
     url: window.location.href + endpoint,
     type: method,
     data: data,
-    success: callback,
-    error: callbackErr
+    success: function(data, status, xhr) {
+      // cleaning error layout before updating layour 
+      $("#error").hide()
+      $("#logged_out").hide();
+
+      // Set Agent state to "connected"
+      $("#agent_status").html("Connected<br>to Agent");
+      $("#agent_status").removeClass("disconnected")
+      $("#agent_status").addClass("connected")
+
+      try {
+        callback(data, status, xhr)
+      } catch (error) {
+        console.log(error)
+      }
+    },
+    error: function(requestObject, error, errorThrown) {
+      try {
+        callbackErr(requestObject, error, errorThrown)
+      } catch (error) {
+        console.log(error)
+      }
+
+      // Set Agent state to "disconnected"
+      $("#agent_status").html("Not connected<br>to Agent");
+      $("#agent_status").removeClass("connected")
+      $("#agent_status").addClass("disconnected")
+
+      // Display error layout 
+      setError(requestObject.status, requestObject.responseText)
+    }
   })
 }
 
+// setError print the 
+/**
+ * Sets the error message and displays it on the page.
+ * @param {number} status - The status code of the error.
+ * @param {string} message - The error message.
+ */
+function setError(status, message) {
+  if (status == 0) {
+    message = "Unable to contact the Datadog Agent. Please ensure it is running."
+  }
+  else if (status == 401) {
+    message = `Not logged in. Please ensure that your GUI session has not expired. (initial message: ${message.trim()})`
```

Comment:
<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏What do you think of eg. `Agent replied with ...` instead of `(initial message: ...)` ?